### PR TITLE
Add embaas document extraction api endpoints 

### PR DESCRIPTION
**Pull request by juliuslipp**
# Introduces embaas document extraction api endpoints

In this PR, we add support for embaas document extraction endpoints to Text Embedding Models (with LLMs, in different PRs coming). We currently offer the MTEB leaderboard top performers, will continue to add top embedding models and soon add support for customers to deploy thier own models. Additional Documentation + Infomation can be found [here](https://embaas.io).

While developing this integration, I closely followed the patterns established by other langchain integrations. Nonetheless, if there are any aspects that require adjustments or if there's a better way to present a new integration, let me know! :)

Additionally, I fixed some docs in the embeddings integration.

Related PR: #5976 

#### Who can review?
  DataLoaders
  - @eyurtsev

### CODE REVIEW

File: docs/modules/indexes/document_loaders/examples/embaas.ipynb [hunk 1]
```diff
@@ -0,0 +1,167 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "source": [
+    "## embaas\n",
```
Comment (rlancemartin) [inline docs/modules/indexes/document_loaders/examples/embaas.ipynb:6]: Please change to header 1 (for documentation): `# embaas`

**Review by rlancemartin — APPROVED** [review]
Thanks for the contribution! Minor nit on notebook header.

File: docs/modules/indexes/document_loaders/examples/embaas.ipynb (hunk 1 above)
Comment (juliuslipp) [reply docs/modules/indexes/document_loaders/examples/embaas.ipynb]: Done :) Changed to upper-case as well. Thanks!

**Review by rlancemartin — APPROVED** [review]
Thanks!

**Pull request merged**